File: frontend/src/constants/courses.js
// Cấp độ khóa học hiển thị ở catalog và trang quản trị.
export const LEVEL_LABELS = {
  beginner: "Cơ bản",
  intermediate: "Trung cấp",
  advanced: "Nâng cao",
};

export const COURSE_STATUS_LABELS = {
  active: "Đang mở",
  inactive: "Tạm ẩn",
};

export const EXAM_BADGES = {
  IELTS: "badge-blue",
  TOEIC: "badge-green",
  OTHER: "badge-gray",
};

export const getExamBadgeClass = (examType) => EXAM_BADGES[examType] || "badge-gray";

File: backend/src/main/java/com/newskyenglish/model/Courses.java
package com.newskyenglish.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "courses")
@Data @NoArgsConstructor @AllArgsConstructor @Builder
// Entity biểu diễn khóa học mà trung tâm đang mở bán hoặc giảng dạy.
public class Courses {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "course_id")
    private Long id;

    @Column(name = "title", length = 150)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "price", precision = 10, scale = 2)
    private BigDecimal price;

    @Enumerated(EnumType.STRING)
    @Column(name = "level")
    private Level level;

    @Enumerated(EnumType.STRING)
    @Column(name = "course_type")
    private ExamType examType;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    @Builder.Default
    private Status status = Status.active;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum Level { beginner, intermediate, advanced }
    public enum Status { active, inactive, archived }
    public enum ExamType { IELTS, TOEIC, OTHER }
}

